tiledb/cf/creator.py

```diff
@@ -1078,17 +1197,22 @@ def __repr__(self):
             for dim_filter in self.filters:
                 filters_str += repr(dim_filter) + ", "
             filters_str += "])"
-        return f"DimCreator({repr(self.base)}, tile={self.tile}{filters_str})"
+        return f"DimCreator({repr(self._base)}, tile={self.tile}{filters_str})"
+
+    @property
+    def base(self) -> SharedDim:
+        """Shared definition for the dimensions name, domain, and dtype."""
+        return self._base
```

Comment:
Why make the attribute "private" if it's going to be exposed anyway?

Reply:
It is only exposed for getting, not setting.